CourseProject/src/main/java/com/neoflex/CourseProject/controller/CalculatorController.java

```diff
@@ -1,36 +1,63 @@
 package com.neoflex.CourseProject.controller;
 
+
 import com.neoflex.CourseProject.dto.CreditDto;
 import com.neoflex.CourseProject.dto.LoanOfferDto;
 import com.neoflex.CourseProject.dto.LoanStatementRequestDto;
 import com.neoflex.CourseProject.dto.ScoringDataDto;
-import com.neoflex.CourseProject.exception.ValidationException;
 import com.neoflex.CourseProject.service.PreScoringService;
+import com.neoflex.CourseProject.service.ScoringService;
 import com.neoflex.CourseProject.service.ValidationService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.*;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
 
 import java.util.List;
 
 @RestController
+@Slf4j
+@Tag(name = "Calculator API")
 public class CalculatorController {
+
+    private final ScoringService scoringServiceImpl;
+    private final PreScoringService preScoringServiceImpl;
+    private final ValidationService validationServiceImpl;
+
     @Autowired
-    PreScoringService preScoringService;
-    @Autowired
-    ValidationService validationService;
+    public CalculatorController(PreScoringService preScoringServiceImpl, ScoringService scoringServiceImpl,
+                                ValidationService validationServiceImpl) {
+        this.preScoringServiceImpl = preScoringServiceImpl;
+        this.scoringServiceImpl = scoringServiceImpl;
+        this.validationServiceImpl = validationServiceImpl;
+    }
 
+    @Operation(
+            summary = "расчёт возможных условий кредита",
+            description = "На основании LoanStatementRequestDto происходит прескоринг, создаётся 4 кредитных предложения LoanOfferDto"
+    )
     @PostMapping("/calculator/offers")
-    public List<LoanOfferDto> calculationOfPossibleLoanTerms(@RequestBody LoanStatementRequestDto loanStatementRequestDto) throws ValidationException {
-        validationService.resultsOfValidation(loanStatementRequestDto);
-        return (preScoringService.resultsOfPreScoring(loanStatementRequestDto.getAmount()));
+    public List<LoanOfferDto> calculationOfPossibleLoanTerms(@RequestBody LoanStatementRequestDto loanStatementRequestDto){
+        log.info("requested loanStatementRequestDto: {}", loanStatementRequestDto);
+        validationServiceImpl.resultsOfValidation(loanStatementRequestDto);
+        log.info("response list of LoanOfferDto: {}", preScoringServiceImpl.resultsOfPreScoring(loanStatementRequestDto.getAmount()));
+        return (preScoringServiceImpl.resultsOfPreScoring(loanStatementRequestDto.getAmount()));
```

Comment:
Хватит и лога на выходные данные на уровне сервиса перед return, либо же, если хочешь сделать это непосредственно в контроллере, то чтобы избавиться от дублирования вызова метода в логе и в return, вынеси в отдельную переменную:
```suggestion
        log.info("requested loanStatementRequestDto: {}", loanStatementRequestDto);
        validationServiceImpl.resultsOfValidation(loanStatementRequestDto);
        CreditDto result = preScoringServiceImpl.resultsOfPreScoring(loanStatementRequestDto.getAmount()
        log.info("response list of LoanOfferDto: {}", result);
        return result;
```

Reply:
исправлено